### PR TITLE
[record-minmax] Add input_data_format option to record-minmax

### PR DESCRIPTION
**Pull request by jinevening**
This adds input_data_format option to record-minmax.

ONE-DCO-1.0-Signed-off-by: Hyukjin Jeong <hj1.jeong@samsung.com>

---
Related to: #7132
Draft PR: #7158

### CODE REVIEW

File: compiler/record-minmax/driver/Driver.cpp [hunk 1]
```diff
@@ -127,8 +136,24 @@ int entry(const int argc, char **argv)
...
     auto input_data_path = arser.get<std::string>("--input_data");
 
-    // Profile min/max while executing the given input data
-    rmm.profileData(mode, input_data_path, min_percentile, max_percentile);
+    if (input_data_format == "h5" || input_data_format == "hdf5")
+    {
+      // Profile min/max while executing the H5 data
+      rmm.profileData(mode, input_data_path, min_percentile, max_percentile);
+    }
+    // input_data is a text file where the absolute path of each record (a representative data) is
```
Comment (seanshpark) [inline compiler/record-minmax/driver/Driver.cpp:144]: > absolute path

Actual code just reads the file so I thin relative to CWD would also work.
If your intention is to use absolute path, them we need to check this.
If not, how about changing this to

> file where the file path of each record (a representative data) is

Comment (seanshpark) [reply compiler/record-minmax/driver/Driver.cpp:144]: > each record

Is this each line of the text file? using word `record` is hard to understand the unit or something...

File: compiler/record-minmax/driver/Driver.cpp [hunk 2]
```diff
@@ -127,8 +136,24 @@ int entry(const int argc, char **argv)
...
+    // written line-by-line. Each record should contain all inputs of a model, concatenated in the
```
Comment (seanshpark) [inline compiler/record-minmax/driver/Driver.cpp:145]: `record` here gives better understanding but still, it's a bit hard to imagine

```
input data is a text file having file path in each line.
each data file is composed of multiple sets of input data of the model,
concatenated  in same order with the input index of the model

[input 1 data][input 2 data]...[input n data] | [input 1 data][input 2 data]...[input n data] ...
|<-----------   set 1    ------------------>|   |<-----------   set 2    ------------------>|
```

Comment (jinevening) [reply compiler/record-minmax/driver/Driver.cpp]: According to the picture, a data file should contain a single set (composed of all inputs of a model).

I've added a picture for easier understading. PTAL.

**Review by seanshpark — APPROVED** [review]
LGTM thank you!

**Pull request merged**